lib/request-wrapper.ts

```diff
@@ -36,10 +37,29 @@ import {
 import logger from './logger';
 import { streamToPromise } from './stream-to-promise';
 
+/**
+ * Retry configuration options.
+ */
+export interface RetryOptions {
+  /**
+   * Maximum retries to attempt.
+   */
+  maxRetries?: number;
+
+  /**
+   * Ceiling for the retry delay (in seconds) - delay will not exceed this value.
+   */
+  maxRetryInterval?: number;
+}
+
 export class RequestWrapper {
-  private axiosInstance;
+  private axiosInstance: AxiosInstance;
+
+  private retryInterceptorId: number;
 
-  private compressRequestData: boolean;
+  private raxConfig: rax.RetryConfig;
+
+  public compressRequestData: boolean;
```

Comment:
I liked your idea of keeping this private and adding a setter, now that we've figured out the "not a function" error

Reply:
Yes, changed it back